### PR TITLE
Multi model

### PR DESCRIPTION
**Pull request by curadomr**
Modifications to implement addMapping functionality, including tests and checks. 
mapping object must be a list, with each element consisting in a vector of strings corresponding to the features found in a model, and named with the same name of the model.

### CODE REVIEW

**Comment by jdblischak** [comment]
@curadomr Thanks for the PR! Overall this looks good. We'll need to remove the app files in `inst/build/` from the commit.

I assume the GitHub Actions jobs aren't running because they are only triggered by a PR to "main", and this was originally opened against "dev".

https://github.com/abbvie-external/OmicNavigator/blob/ca1cbf14d8ed66c47341c5ca2c065ac1f10562b1/.github/workflows/comprehensive.yml#L20-L22

I'm going to close and re-open the PR in an attempt to trigger GitHub Actions 🤞

**Comment by jdblischak** [comment]
The Windows and macOS jobs with R 4.1.2 passed.

The Ubuntu job with R 3.4.4 failed to build the vignettes. It doesn't like this line from `checkMapping()`:

```R
Error : vapply(mappingdf, is.character, logical(1)) are not all TRUE
```

**Comment by jdblischak** [comment]
The vignette builds fine on my local Ubuntu with R 4.1.1. It's likely an issue with the older R version (3.4.4), as opposed to an issue with Ubuntu.

**Comment by jdblischak** [comment]
`R CMD check` passed locally for me both for 1) R 4.0.5 on Windows, and 2) R 4.1.1 on Ubuntu.

File: R/check.R [hunk 1]
```diff
@@ -402,6 +402,39 @@ checkPlots <- function(plots) {
...
+
+  # stop if mapping object has less than 2 elements
+  if (length(mapping) > 0) stopifnot(length(mapping) > 1)
+  else return(NULL)
+
+  # check if list elements have the same size. If not, fill difference with NA.
+  listMaxLength <- max(sapply(mapping, length))
+  mapping <- lapply(lapply(mapping, unlist), "length<-", listMaxLength)
+
+  mappingdf <- as.data.frame(mapping)
```
Comment (jdblischak) [inline R/check.R:416]: I figured out the error that only occurs with R 3.4.4. In R3, strings are always automatically coerced to factors by default (`stringsAsFactors=TRUE`). This default behavior was changed (for the better) in R4. Please update this line to explicitly keep the strings as charactor vectors when converting to a data frame.

```R
 mappingdf <- as.data.frame(mapping, stringsAsFactors = FALSE)
```

File: R/check.R [hunk 2]
```diff
@@ -402,6 +402,39 @@ checkPlots <- function(plots) {
...
+  # check if any given model has at least one feature aligned with another model
+  for (i in seq_along(mappingdf)) {
+    tempModel   <- mappingdf[!is.na(mappingdf[[i]]),]
+    if (nrow(tempModel) > 1) {
+      featAligned <- any(apply(!sapply(tempModel, is.na), 1, sum) > 1)
+    } else {
+      featAligned <- any(sum(!sapply(tempModel, is.na)) > 1)
+    }
+
+    if (isFALSE(featAligned)) {
```
[the remark this answers is not in the document]
Comment (curadomr) [reply R/check.R]: Just implemented in most recent commit. Thanks!

**Review by jdblischak — APPROVED** [review]

**Pull request merged**